sdk/core/azure-core/src/main/java/com/azure/core/http/ResponseStatusOption.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http;
+
+/**
+ * ResponseStatusOption controls the behavior of an operation based on the status code of a response.
+ */
+public enum ResponseStatusOption {
+    /**
+     * Indicates that an operation should throw an exception when the response indicates a failure.
+     */
+    DEFAULT,
```

Comment:
DEFAULT doesn't indicate what is the expected behavior. Instead, we should name this as `THROW_EXCEPTION`  and document that it is the default behavior.

Reply:
This is the name used in .NET - but yeah I agree we could use a better name like `THROW_400_AND_ABOVE`

Reply:
Check with Matt on the .NET design

Reply:
Changed to `.setThrowOnError(boolean)`.